### PR TITLE
Better email failure handling in hermes

### PR DESCRIPTION
**Pull request by brianlovin**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hermes

cc @mxstbr - main review points:
- I'm adding a db connection in hermes. Any concerns about this? Would rather have direct access to turn off notification settings directly in hermes rather than creating a new job for some other worker to handle
- could you double check my syntax in the `send-email.js` file for handling the error?

Also - this will *not* remove the failed job from Hermes - that should be a separate task for #2768 probably. This will however reduce the volume of failures as we slowly start to deactivate users notification settings when they generate a hard bounce or spam complaint

### CODE REVIEW

File: hermes/models/db.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const config = IS_PROD
+  ? {
+      ...DEFAULT_CONFIG,
+      ...PRODUCTION_CONFIG,
+    }
+  : {
+      ...DEFAULT_CONFIG,
+    };
+
+var r = require('rethinkdbdash')(config);
```
Comment (dev-drprasad) [inline hermes/models/db.js]: @brianlovin why you used `var` instead of `const` ? is there any special reason ?

File: hermes/send-email.js [hunk 2]
```diff
@@ -41,11 +44,21 @@ const sendEmail = (options: Options) => {
...
-          return rej(err);
+          console.error('Error sending email:');
+          console.error(err);
+
+          // 406 means the user became inactive, either by having an email
+          // hard bounce or they marked as as spam
+          if (err.code === 406) {
+            console.error(`Deactivating future notifications for ${To}`);
+            return await deactiveUserEmailNotifications(To)
+              .then(() => rej(err))
```
Comment (mxstbr) [inline hermes/send-email.js]: Why are we `console.error`ing here if we know what the problem is anyway? This code should read:

```JS
async err => {
  if (err.code === 406) {
    return deactiveUserEmailNotifications(To)
      .catch(err => rej(err))
  }
  
  console.error(`Error sending email:`, err);
}
```

Sentry picks up `console.error`s, and we don't really want to get a log of an error we handled properly anyway, right?

File: hermes/send-email.js [hunk 3]
```diff
@@ -41,11 +44,21 @@ const sendEmail = (options: Options) => {
...
+      async err => {
         if (err) {
-          console.log('Error sending email:');
-          console.log(err);
-          return rej(err);
+          console.error('Error sending email:');
+          console.error(err);
+
+          // 406 means the user became inactive, either by having an email
+          // hard bounce or they marked as as spam
```
Comment (dev-drprasad) [inline hermes/send-email.js]: small typo error. there are two **as**

**Comment by brianlovin** [comment]
Those fixes have been applied! Thanks team :)

File: hermes/send-email.js (hunk 2 above)
Comment (brianlovin) [reply hermes/send-email.js]: Why would sentry pick up `console.error`s? Regardless, I removed it just in case :)

**Pull request merged**